File: judo-data/src/main/java/com/judopay/customer/Country.java
package com.judopay.customer;

public class Country {

    public static final String UNITED_KINGDOM = "UK";
    public static final String UNITED_STATES = "USA";
    public static final String CANADA = "Canada";
    public static final String OTHER = "Other";

    private final int code;
    private final String displayName;

    public Country(int code, String displayName) {
        this.code = code;
        this.displayName = displayName;
    }

    public int getCode() {
        return code;
    }

    public String getDisplayName() {
        return displayName;
    }

}

File: judopay-sdk/src/main/java/com/judopay/model/Country.kt
package com.judopay.model

// ISO 3166-2 list of supported card form countries
enum class Country {
    GB,
    US,
    CA,
    OTHER
}

fun String.asCountry() = Country.values().firstOrNull { it.displayName.equals(this, true) }

val Country.displayName: String
    get() = when (this) {
        Country.GB -> "UK"
        Country.US -> "USA"
        Country.CA -> "Canada"
        Country.OTHER -> "Other"
    }

val Country.postcodeMaxLength: Int
    get() = when (this) {
        Country.US -> 5
        Country.CA -> 6
        Country.GB,
        Country.OTHER -> 8
    }
